cilium/cmd/config.go

```diff
@@ -70,20 +76,55 @@ func configDaemon(cmd *cobra.Command, opts []string) {
 		}
 	} else if len(opts) == 0 {
```

Comment:
The contents of this else if block has gotten a lot more complex. I think it would be worth moving it to its own dedicated function.